### PR TITLE
Add reducer and API function for contributors

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #864 

#### What's this PR do?
Adds reducer and fetch function for contributors

#### How should this be manually tested?
There's no code using this yet

### CODE REVIEW

File: static/js/factories/channels.js [hunk 1]
```diff
@@ -25,11 +32,38 @@ export const makeChannelList = (numChannels: number = 20) => {
...
-export const makeModerators = (username: ?string = null) =>
-  username
-    ? [
-      {
-        moderator_name: username
-      }
-    ]
-    : []
+export const makeContributor = (username: ?string = null): Contributor => ({
+  contributor_name: username ? username : casual.word,
```
Comment (aliceriot) [inline static/js/factories/channels.js]: I think this could just be `username || casual.word`

File: static/js/factories/channels.js [hunk 2]
```diff
@@ -25,11 +32,38 @@ export const makeChannelList = (numChannels: number = 20) => {
...
+  makeContributor(username),
+  makeContributor(null),
+  makeContributor(null)
+]
+
+export const makeModerator = (
+  username: ?string = null,
+  isModerator: ?boolean
+): Moderator => ({
+  moderator_name: username ? username : casual.word,
```
Comment (aliceriot) [inline static/js/factories/channels.js]: same comment here

File: static/js/flow/discussionTypes.js [hunk 3]
```diff
@@ -146,10 +146,20 @@ export type ReplaceMoreCommentsPayload = {
 
 export type Moderator = {
   moderator_name: string,
+  email?: string,
+  full_name?: string,
```
Comment (aliceriot) [inline static/js/flow/discussionTypes.js]: what does it signify if these two fields are included or not?

File: static/js/reducers/channel_contributors.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// @flow
+import { GET, INITIAL_STATE } from "redux-hammock/constants"
+
+import type { ChannelContributors } from "../flow/discussionTypes"
```
Comment (aliceriot) [inline static/js/reducers/channel_contributors.js]: small thing, but can you move the type import below the `../lib/api` import (and put a blank line in-between)

**Comment by noisecapella** [comment]
Comments addressed

**Pull request merged**